Source/Scene/ShadowMap.js

```diff
@@ -1211,16 +1233,90 @@ define([
         }
     }
 
+    var scratchCartesian1 = new Cartesian3();
+    var scratchCartesian2 = new Cartesian3();
+    var scratchBoundingSphere = new BoundingSphere();
+    var scratchCenter = scratchBoundingSphere.center;
+
+    function checkVisibility(shadowMap, frameState) {
+        var camera = frameState.camera;
+        var lightCamera = shadowMap._lightCamera;
+        var sceneCamera = shadowMap._sceneCamera;
+        var shadowMapCamera = shadowMap._shadowMapCamera;
+
+        var boundingSphere = scratchBoundingSphere;
+
+        // Check whether the shadow map is in view and needs to be updated
+        if (shadowMap._cascadesEnabled) {
+            // If the nearest shadow receiver is further than the shadow map's maximum distance then the shadow map is out of view.
+            if (sceneCamera.frustum.near > shadowMap._maximumDistance) {
+                shadowMap._outOfView = true;
+                shadowMap._needsUpdate = false;
+                return;
+            }
+
+            // If the light source is below the horizon then the shadow map is out of view
+            var surfaceNormal = frameState.mapProjection.ellipsoid.geodeticSurfaceNormal(sceneCamera.positionWC, scratchCartesian1);
+            var lightDirection = Cartesian3.negate(shadowMapCamera.directionWC, scratchCartesian2);
+            var dot = Cartesian3.dot(surfaceNormal, lightDirection);
+            if (dot < 0.1) {
+                shadowMap._outOfView = true;
+                shadowMap._needsUpdate = false;
+                return;
+            }
+
+            // If the scene camera or light direction changes, then the shadow map needs to update
+            var sceneCameraChanged = !Matrix4.equals(sceneCamera.viewMatrix, camera.viewMatrix);
+            var directionChanged = !Cartesian3.equals(shadowMapCamera.directionWC, lightCamera.directionWC);
+            shadowMap._needsUpdate = sceneCameraChanged || directionChanged;
+
+            // By default cascaded shadows are always in view
+            shadowMap._outOfView = false;
+        } else if (shadowMap._isPointLight) {
+            // Sphere-frustum intersection test
+            boundingSphere.center = shadowMapCamera.positionWC;
+            boundingSphere.radius = shadowMap._radius;
+            shadowMap._outOfView = frameState.cullingVolume.computeVisibility(boundingSphere) === Intersect.OUTSIDE;
+            shadowMap._needsUpdate = !shadowMap._outOfView && !shadowMap._boundingSphere.equals(boundingSphere);
+            BoundingSphere.clone(boundingSphere, shadowMap._boundingSphere);
+        } else {
+            // Simplify frustum-frustum intersection test as a sphere-frustum test
```

Comment:
Is it worth adding a more precise test on the roadmap?  Or perhaps the CPU overhead will be too high to justify it?

BTW have you seen http://www.realtimerendering.com/intersections.html?